test/unit/migration_config_test.py

```diff
@@ -25,3 +26,21 @@ def test_get_migration_product_targets(self, mock_error):
         with raises(DistMigrationProductNotFoundException):
             self.config.get_migration_product()
             assert mock_error.called
+
+    @patch('suse_migration_services.logger.log.info')
+    @patch('yaml.dump')
+    @patch.object(Defaults, 'get_migration_config_file')
+    @patch.object(Defaults, 'get_system_migration_custom_config_file')
+    def test_update_migration_config_file(
+        self, mock_get_system_migration_config_custom_file,
+        mock_get_migration_config_file, mock_yaml_dump, mock_info
+    ):
+        mock_get_system_migration_config_custom_file.return_value = \
+            '../data/optional-migration-config.yml'
+        mock_get_migration_config_file.return_value = \
+            '../data/combined-migration-config.yml'
+
+        self.config.update_migration_config_file()
+        assert self.config.get_migration_product() == 'SLES/15.1/x86_64'
+        assert mock_info.called
+        os.remove('../data/combined-migration-config.yml')
```

Comment:
This test has room for improvement. The code actually writes the file and your test re-reads it and after that deletes the file. You expect the system that runs your tests to have permissions to write on the target storage. This is an assumption which could be wrong especially for running unit tests. So normally you would mock the open call and perform any read/write check in memory. See units/kernel_load_test.py or fstab_test.py if you need examples how to mock code that opens and write files.

We can go with this version but it's not so nice :)